plugins/nginx-vhosts/functions

```diff
@@ -93,147 +35,158 @@ restart_nginx() {
   esac
 }
 
-nginx_build_config() {
-  local APP="$1"; local DOKKU_APP_LISTEN_PORT="$2"; local DOKKU_APP_LISTEN_IP="$3"
-  verify_app_name "$APP"
-  VHOST_PATH="$DOKKU_ROOT/$APP/VHOST"
-  URLS_PATH="$DOKKU_ROOT/$APP/URLS"
-  WILDCARD_SSL_PATH="$DOKKU_ROOT/tls"
-  APP_SSL_PATH="$DOKKU_ROOT/$APP/tls"
-  APP_NGINX_TEMPLATE="$DOKKU_ROOT/$APP/nginx.conf.template"
-  APP_NGINX_SSL_TEMPLATE="$DOKKU_ROOT/$APP/nginx.ssl.conf.template"
-
-  eval "$(config_export global)"
-  eval "$(config_export app $APP)"
-
-  if [[ ! -n "$DOKKU_NO_NGINX" ]]; then
-    if [[ -z "$DOKKU_APP_LISTEN_PORT" ]] && [[ -z "$DOKKU_APP_LISTEN_IP" ]]; then
-      shopt -s nullglob
-      for DOKKU_APP_IP_FILE in $DOKKU_ROOT/$APP/IP.web.*; do
-        DOKKU_APP_PORT_FILE="${DOKKU_APP_IP_FILE//IP/PORT}"
-        DOKKU_APP_LISTENER_IP=$(< $DOKKU_APP_IP_FILE)
-        DOKKU_APP_LISTENER_PORT=$(< $DOKKU_APP_PORT_FILE)
-
-        DOKKU_APP_LISTENERS+=" "
-        DOKKU_APP_LISTENERS+="$DOKKU_APP_LISTENER_IP:$DOKKU_APP_LISTENER_PORT"
-        DOKKU_APP_LISTENERS+=" "
-      done
-      shopt -u nullglob
-    fi
+nginx_logs() {
+  local desc="display app nginx logs"
+  local APP="$2"; verify_app_name "$APP"
+  local NGINX_LOGS_TYPE=${1#nginx:}
+  local NGINX_LOGS_TYPE=${NGINX_LOGS_TYPE%-logs}
+  local NGINX_LOGS_PATH="/var/log/nginx/$APP-$NGINX_LOGS_TYPE.log"
 
-    DOKKU_APP_CIDS=($(get_app_container_ids $APP))
-    for file in nginx.conf.template nginx.ssl.conf.template; do
-      docker cp "${DOKKU_APP_CIDS[0]}:/app/${file}" "$DOKKU_ROOT/$APP/" 2> /dev/null || true
-    done
-
-    [[ -f "$APP_NGINX_TEMPLATE" ]] && NGINX_TEMPLATE="$APP_NGINX_TEMPLATE" && NGINX_CUSTOM_TEMPLATE="true" && dokku_log_info1 'Overriding default nginx.conf with detected nginx.conf.template'
-
-    NGINX_CONF=$(mktemp -t "nginx.conf.XXXXXX")
-    SCHEME="http"
+  if [[ $3 == "-t" ]]; then
+    NGINX_LOGS_ARGS="-F"
+  else
+    NGINX_LOGS_ARGS="-n 20"
+  fi
 
-    # if the app has is not vhost enabled then, let's make sure we're cleaned up
-    [[ "$(is_app_vhost_enabled $APP)" == "false" ]] && disable_app_vhost $APP --no-restart
+  tail "$NGINX_LOGS_ARGS" "$NGINX_LOGS_PATH"
+}
 
-    if [[ -z "$DOKKU_NGINX_PORT" ]]; then
-      if [[ "$(is_app_vhost_enabled $APP)" == "false" ]]; then
+configure_nginx_ports() {
+  local desc="configure nginx listening ports"
+  local APP=$1; verify_app_name "$APP"
+  local RAW_TCP_PORTS="$(get_app_raw_tcp_ports $APP)"
```

Comment:
My thinking is any var that gets set as a config variable should be prefixed but as a local var in a function, it's not necessary